### ISSUE TITLE
Support multiple controllers on Android

### ISSUE THREAD

**Issue by msangel** [stenzek/duckstation#1787]
**Game details**
any

**Description of the issue / bug**
This bug is about android version of the application.
There are two controllers connected in HID mode but the app cant see them as different. Both controllers trigger actions as the first one, and so the second is not visible as separate.  Both controllers have the same name as basically is the same model.

** Device **
host: Default android 10 device with stock firmware and without rooting. Exact android version:  10 (QP1A.190711.020)
gamepad: generic PS3-compatible with Bluetooth HID interface, model GamePro MG550
No memory optimizers installed.
The bug reproduced in any GPU Renderer.

**Steps to reproduce**
1) Connect first gamepad as  MG550
2) Connect second gamepad as MG550 (note: setting different names in android Bluetooth device manager is not affecting these names)
3) Run app
4) The app see both controllers both works as single first

**Emulator version**
DuckStation for Android 0.1-3431-gc3d2999d (dev)
Built Mar  4 2021 15:25:33

**Additional context**
The tester apps and ePSXe can see that as separate controllers. Screens below.
The android settings:
<img src="https://user-images.githubusercontent.com/773163/110737374-450a2800-81fb-11eb-95e4-f229d618124a.png" width="45%" height="45%"> 
The duckstation settings:
<img src="https://user-images.githubusercontent.com/773163/110737385-49cedc00-81fb-11eb-88f4-18ce6afd8403.png " width="45%" height="45%"> | <img src="https://user-images.githubusercontent.com/773163/110737396-4d626300-81fb-11eb-840e-37cb481d5c05.png" width="45%" height="45%"> 
The ePSXe settings:
<img src="https://user-images.githubusercontent.com/773163/110737412-55220780-81fb-11eb-9cf1-c19a690c742a.png" width="32%" height="32%"> | <img src="https://user-images.githubusercontent.com/773163/110737401-505d5380-81fb-11eb-9b82-25125a1b417b.png" width="32%" height="32%"> | <img src="https://user-images.githubusercontent.com/773163/110737408-52bfad80-81fb-11eb-8322-183816e3b4aa.png" width="32%" height="32%">

Important note: there is a bug in ePSXe there: on controller reconnect the app should be restarted(and the controller turn itself after 5 minutes of not using, and this causes pain).  So be ready that the controller may be disconnected and then reconnected at any time of the game.

**Comment by stenzek** [comment]
Only one controller is supported on Android. I don't really have much motivation to add it, but I'll leave the issue open anyway.

**Comment by stenzek** [comment]
(My main concern would be supporting multiple controllers would require rebinding controls to new values, and a good chunk of the userbase acted as if I'd personally attacked them when they had to bind controllers. If they have to rebind it again, they'll be really angry.... :/)

**Comment by msangel** [comment]
In the ePSXe the real id is taking via 
```
InputDevice device = event.getDevice();
String full = device.toString();
int in = full.indexOf("Descriptor:");
if (in == -1) {
    // fallback
    String name = device.getName();
    int id = event.getDeviceId();
    return "###" + name + "###" + id + "###";
}
String descriptor = full.substring(in + 12, full.indexOf(10, in + 12));
```

Of course the code above is reproduced based on assumptions as the ePSXe is a closed source :smirk:

**Comment by stenzek** [comment]
I'd probably just use the descriptor, as it's supposed to be stable: https://developer.android.com/reference/android/view/InputDevice#getDescriptor()

**Comment by stenzek** [comment]
Implemented in PR #1806, will push to preview soon.
